src/client_side.cc

```diff
@@ -3859,6 +3897,7 @@ ConnStateData::borrowPinnedConnection(HttpRequest *request, const AccessLogEntry
     if (pinning.peerAccessDenied)
         throw pinningError(ERR_CANNOT_FORWARD); // or generalize ERR_FORWARDING_DENIED
 
+    pinning.borrowed = true;
```

Comment:
This method already handles cache_peers that are gone - after the above toGoneCachePeer() check the connection will be closed in this case. So it cannot borrow pinned connection to the cache_peer that has been just reconfigured away. I thought that this was possible in the old code - but my previous test was incorrect - it lacked 'never_direct allow all'  setting. I think we are going to maintain the old behavior, and thus can remove the pinning.borrowed flag. We may also need to check that the cache_peer is gone while the pinned connection was busy just before making the connection idle, e.g., adding another toGoneCachePeer() check into notePinnedConnectionBecameIdle().